### ISSUE TITLE
[Gateways] Large json response malformed in gravitee 3.20.1

### ISSUE THREAD

**Issue by fabMrc** [gravitee-io/issues#8907]
I used 3.17 gravitee gateway version and I have just upgraded to 3.20.1\. Now we can see on a specific api the response content is malformed . It was not the case before. I tried to analyse it :

*   Backend service direct request (bypass gravitee) : OK
*   Direct gateway request (bypass our F5 BIGIP that manage gateway pool): KO
*   Classic request (using our full stack): KO

The json weigth is 4888ko and the content malformed is for exemple <g-emoji class="g-emoji" alias="+1" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png">
 👍 
 </g-emoji> 

 { "type":"dictionaryValue", "id":null, "code":"M", "value":null, "operator":"=", "description":"Mois ", "scoped_out":"nul""uniqueIdentifier":"62842399null" } 

      

or

}, "isDiscriminating":false, "weighting":1.0, "rate":0.0, "validValues""alse", "discriminating":false },

I desactivated gzip compression on api config because I know http server can compress content if too heavy but it is the same error

I am available for further analysis

**Comment by fabMrc** [comment]
I downgrade to 3.19.0 no error. I am going to proceed by dychotomy to isolate the version where code change lead to the gateway bug

**Comment by brasseld** [comment]
Hey @fabMrc 

Are you using some specific policies on the response ?

**Comment by fabMrc** [comment]
no policies and no plan juste un free plan

**Comment by fabMrc** [comment]
Ok In 3.19.8 the error is reproducible so an impact is done by a code change between 3.19.0 and 3.19.8

**Comment by brasseld** [comment]
Is it just an issue with formatting, or the content is also altered in some ways ?

**Comment by fabMrc** [comment]
no it is the response content that is altered

**Comment by fabMrc** [comment]
Yes of course by mail

**Comment by brasseld** [comment]
Backend response headers:

```
Expires: 0
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
X-XSS-Protection: 1; mode=block
Pragma: no-cache
X-Frame-Options: DENY
Date: Mon, 27 Feb 2023 13:23:32 GMT-1s
Connection: keep-alive
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
Transfer-Encoding: chunked
Content-Type: application/warm.warranty-1.0+json;charset=UTF-8
```

**Comment by jhaeyaert** [comment]
Hi @fabMrc, can you provide more information regarding the way the gateway & your api are configured? Ex:

- Does the gateway enable TLS?
- Did you enable alpn on the gateway?
- Does your backend expose in TLS? 
- What is the protocol exposed by your backend (http1.1, http2)?
- Does your backend chunk the body response (seems it is)? If so, are you able to make a test by changing the way the backend returns the body response, switching to Content-Length instead of chunked?
- Did you enable logging of headers and body on your API?
- Did you notice the issue on all paths & will all methods of your API? For instance, do you see an issue when you POST but not when you GET?
- Did you notice any difference when you are calling the gateway with http1.1 and http2 protocols?
- What is the topology of your installation? Do you have any reverse proxy or something between your gateway and your backend (ex: nginx, traefik, istio, ...)?

Finally, are you able to provide a reproducible case? Whatever configuration we try, we are unable to reproduce the issue for now. 

Any other details regarding your context would be really useful.

**Comment by fabMrc** [comment]
• Does the gateway enable TLS? Gateway is exposed with TLS but there is no mTLS enabled	
• Did you enable alpn on the gateway? No
• Does your backend expose in TLS?  Yes (Https)
• What is the protocol exposed by your backend (http1.1, http2)? http1.1
• Does your backend chunk the body response (seems it is)? If so, are you able to make a test by changing the way the backend returns the body response, switching to Content-Length instead of chunked? We are using the default configuration of sprintBoot for restController
•  Did you enable logging of headers and body on your API? Yes and No we tried the two ways
• Did you notice the issue on all paths & will all methods of your API? For instance, do you see an issue when you POST but not when you GET? No just for the method POST in one Backend (large response)
• Did you notice any difference when you are calling the gateway with http1.1 and http2 protocols? We don’t try it
• What is the topology of your installation? Do you have any reverse proxy or something between your gateway and your backend (ex: nginx, traefik, istio, ...)?No the end point is configured directly in the proxy of gravitee-io

Yes it is reproducible every requests is altered

**Comment by fabMrc** [comment]
I will give you the mock next week

**Comment by gaetanmaisse** [comment]
Ok, @jhaeyaert might have found the root cause, I've created a dedicated issue: https://github.com/gravitee-io/issues/issues/8956

**Comment by gaetanmaisse** [comment]
@fabMrc, I will close this issue as we've released APIM 3.20.3, which includes the fix for the issue mentioned in my previous message. 

Can you try it and let us know if it fixes your problem? If not please reopen the issue 😉
